master/custom/steps.py

```diff
@@ -106,3 +107,16 @@ class Uninstall(ShellCommand):
     command = "chmod -R +w target/ &&  rm -rf target/"
     alwaysRun = True
     usePTY = False
+
+class UploadTestResults(steps.FileUpload):
+    warnOnFailure = True
+    haltOnFailure = False
+    flunkOnFailure = False
+    alwaysRun = True
+
+    def __init__(self):
+        super().__init__(
+                workersrc="test-results.xml",
+                masterdest=util.Interpolate("~/test_results/%(prop:buildername)s_%(prop:buildnumber)s.xml")
```

Comment:
Should we make this a publicly-served directory?

Reply:
> Should we make this a publicly-served directory?

What is the best directory that we should use for making it publicly-served?

Reply:
How about `/data/www/buildbot/test-results/`?  I just created it; it [seems to work](https://buildbot.python.org/test-results/).  Daily `.dmg` builds used to be uploaded to `/data/www/buildbot/daily-dmg/`.

Reply:
Updated the PR!